base/linalg/dense.jl

```diff
@@ -74,10 +71,8 @@ julia> isposdef(A)
 true
 ```
 """
-function isposdef(A::AbstractMatrix{T}) where T
-    S = typeof(sqrt(one(T)))
-    isposdef!(S == T ? copy(A) : convert(AbstractMatrix{S}, A))
-end
+isposdef(A::AbstractMatrix{T}) where {T} =
+    isposdef!(copy_oftype(A, promote_type(typeof(chol(one(T))),Float32)))
```

Comment:
So with `isposdef2` as in the comment above we have the following:
```julia
julia> for n in (10, 100, 1000)
           A = rand(n,n); B = A'A
           println("non-hermitian $n x $n")
           @btime isposdef($A)
           @btime isposdef2($A)
           println("hermitian $n x $n")
           @btime isposdef($B)
           @btime isposdef2($B)
       end
non-hermitian 10 x 10
  69.209 ns (2 allocations: 912 bytes)
  7.838 ns (0 allocations: 0 bytes)
hermitian 10 x 10
  487.292 ns (8 allocations: 1.05 KiB)
  500.427 ns (9 allocations: 1.08 KiB)
non-hermitian 100 x 100
  4.054 μs (3 allocations: 78.22 KiB)
  7.653 ns (0 allocations: 0 bytes)
hermitian 100 x 100
  37.041 μs (9 allocations: 78.38 KiB)
  36.770 μs (10 allocations: 78.41 KiB)
non-hermitian 1000 x 1000
  808.359 μs (3 allocations: 7.63 MiB)
  7.651 ns (0 allocations: 0 bytes)
hermitian 1000 x 1000
  6.166 ms (9 allocations: 7.63 MiB)
  6.129 ms (10 allocations: 7.63 MiB)
```

So I will definitely update to the definition in the comment above.